appinventor/appengine/src/com/google/appinventor/server/UploadServlet.java

```diff
@@ -108,6 +109,12 @@ public void doPost(HttpServletRequest req, HttpServletResponse resp) {
           uploadResponse = e.uploadResponse;
         }
       } else if (uploadKind.equals(ServerLayout.UPLOAD_FILE)) {
+        try{
+          URI uri2 = new URI(uri);
+          uri = uri2.getPath();
+        } catch(Exception e) {
+          throw CrashReport.createAndLogError(LOG, req, null, e);
+        }
```

Comment:
Rather than constructing a new URI object, it might make more sense to use [URLDecoder](https://docs.oracle.com/javase/7/docs/api/java/net/URLDecoder.html) to process the percent-encoded URL.